builtin/credential/aws-ec2/backend_test.go

```diff
@@ -1126,6 +1131,7 @@ func TestBackendAcc_LoginAndWhitelistIdentity(t *testing.T) {
 		"policies":     "root",
 		"max_ttl":      "120s",
 		"bound_ami_id": "wrong_ami_id",
+		"bound_iam_role_arn" : "wront_iam_role_arn",
```

Comment:
The login attempt with this data will fail as expected, not because of `wrong_iam_role_arm`, but because of `wrong_ami_id`.

After the login request with this data is complete, role needs to be updated again with correct AMI ID and wrong ARN, and a login request should be initiated again, with an expected failure.

Let me know if this makes sense to you.